src/Algebra/Solver/Ring/WeaklyDecidable.agda

```diff
@@ -0,0 +1,22 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- A weaker form of decidability
+------------------------------------------------------------------------
+
+module Algebra.Solver.Ring.WeaklyDecidable where
+
+open import Relation.Binary.Core using (REL; Decidable)
+open import Relation.Nullary using (Dec; yes; no)
+open import Relation.Unary using (Pred; _⊆_)
+open import Data.Maybe.Base using (Maybe; just; nothing)
+
+module _ {a b ℓ} {A : Set a} {B : Set b} where
+
+  WeaklyDecidable : Pred (REL A B ℓ) _
+  WeaklyDecidable _∼_ = ∀ x y → Maybe (x ∼ y)
+
+  fromDecidable : Decidable ⊆ WeaklyDecidable
+  fromDecidable dec x y with dec x y
+  ... | yes p = just p
+  ... | no _ = nothing
```

Comment:
These definitions should probably respectively go in:
*  `Relation.Binary.Core` for `WeaklyDecidable`
* `Relation.Binary.Consequences` for `fromDecidable`

Reply:
Moving `WeaklyDecidable` to `Relation.Binary.Core` adds an import to `Maybe` which introduces a cyclic dependency between `Relation.Binary.Core`, `Data.Maybe.Base` and `Data.Bool.Base`. In `Data.Bool.Base`, binary relations appear to be used only for `_≟_`, so I will try to move that to `Data.Bool.Properties` in order to resolve the cycle, similarly to what was done for `Data.Nat` recently.